infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/GRPCLogReplicationClientChannelAdapter.java

```diff
@@ -141,7 +143,7 @@ public void send(@Nonnull String nodeId, @Nonnull RequestMsg request) {
     private void queryLeadership(String nodeId, RequestMsg request) {
         try {
             if (blockingStubMap.containsKey(nodeId)) {
-                ResponseMsg response = blockingStubMap.get(nodeId).withWaitForReady().queryLeadership(request);
+                ResponseMsg response = blockingStubMap.get(nodeId).withDeadlineAfter(10, TimeUnit.SECONDS).queryLeadership(request);
```

Comment:
is  `waitDeadlineAfter` with 10 second timeout the same as `withWaitForRead()` ?

Reply:
Hi Sundar, 
`waitForReady` was failing silently as it was retrying for a longer duration, making it harder to debug, `waitDeadlineAfter` throws a visible exception in the logs.
And they recommend this - https://grpc.io/blog/deadlines/